src/test/java/seedu/linkedout/model/applicant/StageTest.java

```diff
@@ -0,0 +1,36 @@
+package seedu.linkedout.model.applicant;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.linkedout.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class StageTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new seedu.linkedout.model.applicant.Stage(null));
+    }
+
+    @Test
+    public void constructor_invalidStage_throwsIllegalArgumentException() {
+        String invalidStage = "";
+        assertThrows(IllegalArgumentException.class, () -> new seedu.linkedout.model.applicant.Stage(invalidStage));
+    }
+
+    @Test
+    public void isValidStage() {
+        // null linkedout
+        assertThrows(NullPointerException.class, () -> seedu.linkedout.model.applicant.Stage.isValidStage(null));
+
+        // invalid stagees
```

Comment:
Same suggestion for the comment directly below

```suggestion
        // invalid stages
```

Reply:
a972860c86657043b96a8bc0fbd3c3823566ffc4